src/Components/Navbar.jsx

```diff
@@ -0,0 +1,145 @@
+import { NavLink, useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { logout } from '../slices/authSlice';
+import navbarData from '../utils/navbarData.json';
+import { ADMIN_ID ,HR_ID} from '../utils/constantData';
+
+const Navbar = () => {
+  const [showLogoutDropdown, setShowLogoutDropdown] = useState(false);
+  const [showNavbarMenu, setShowNavbarMenu] = useState(false);
+  const dropdownRef = useRef(null);
+  const user = useSelector((state) => state.auth.user);
+  const isAuthenticated = useSelector((state) => state.auth.isAuthenticated);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const onLogoutClick = () => {
+    dispatch(logout());
+    navigate('/');
+  };
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
+        setShowLogoutDropdown(false);
+      }
+    };
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  return isAuthenticated && user ? (
+    <nav className="bg-gradient-to-r from-sky-400 to-blue-400 text-white fixed w-full top-0 z-20 sm:px-4">
+      <div className="flex justify-between items-center xl:px-4 sm:py-2 sm:px-3  px-2" >
+        {/* Left Section */}
+        <div className="flex items-center justify-between sm:py-4 py-3 px-1">
+          {/* Mobile Menu Icon */}
+          <FaBars
+            onClick={() => setShowNavbarMenu((prev) => !prev)}
+            className="text-xl md:hidden block cursor-pointer"
+          />
+          {/* Dropdown Menu for Mobile */}
+          <ul
+            className={`${
+              showNavbarMenu ? 'translate-x-0' : '-translate-x-full'
+            } fixed top-12  left-0 h-full bg-gradient-to-r from-sky-400 to-blue-400 space-y-4 px-4 py-2 w-[70vw] transition-transform duration-300 ease-in-out md:hidden`}
+          >
+            <li>
+              <NavLink
+                to="/polls"
+                className={({ isActive }) =>
+                  isActive ? 'text-white' : 'text-black'
+                }
+              >
+                Polls
+              </NavLink>
+            </li>
+            {(user.roleId === ADMIN_ID || user.roleId === HR_ID) &&
+              navbarData.map((item, index) => (
+                <li key={index}>
+                  <NavLink
+                    to={item.path}
+                    className={({ isActive }) =>
+                      isActive ? 'text-white' : 'text-black'
+                    }
+                  >
+                    {item.name}
+                  </NavLink>
+                </li>
+              ))}
+          </ul>
+
+          {/* Horizontal Navbar for Desktop */}
+          <ul className="hidden md:flex md:space-x-6 md:bg-transparent">
+            <li>
+              <NavLink
```

Comment:
remove this navlink use it from navbarData @Himanshusingh770

Reply:
ok sir